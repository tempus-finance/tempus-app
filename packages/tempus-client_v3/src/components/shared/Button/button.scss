.tc__btn {
  padding: 0;
  border: none;
  background-color: transparent;
  cursor: pointer;

  &.tc__btn__primary {
    background-color: var(--primaryMain);
    color: var(--textInverted);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.08);

    &:hover {
      background-color: var(--primaryLight);
      box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.24);
    }
  }

  &.tc__btn__secondary {
    background-color: var(--secondaryMain);
    color: var(--textInverted);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.08);

    &:hover {
      background-color: var(--secondaryLight);
      box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.24);
    }
  }

  &.tc__btn__small {
    width: 96px;
    height: 32px;
    border-radius: 5px;
  }

  &.tc__btn__large {
    width: 144px;
    height: 40px;
    border-radius: 4px;
  }
}
