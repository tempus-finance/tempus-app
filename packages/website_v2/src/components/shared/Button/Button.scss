.tw__button__container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 14px 30px;
  border-radius: 60px;
  border: 1px solid #EEECFF;
  background-color: transparent;
  cursor: pointer;
  box-sizing: border-box;
  width: 283px;
  height: 60px;

  &:hover {
    background-color: #EEECFF;
    color: #050a4a;
  }

  a {
    text-decoration: none;
    display: flex;
    align-items: center;
    color: #EEECFF;
  }
}
