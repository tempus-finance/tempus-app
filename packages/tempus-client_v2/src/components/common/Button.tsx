import { ButtonHTMLAttributes } from 'react';

import './button.scss';

const Button = (props: React.HTMLProps<HTMLButtonElement>) => {
  const { children, ...btnProps } = props;
  const attr = btnProps as ButtonHTMLAttributes<HTMLButtonElement>;

  return (
    <button className="tc__btn" {...attr}>
      {children}
    </button>
  );
};

export default Button;
