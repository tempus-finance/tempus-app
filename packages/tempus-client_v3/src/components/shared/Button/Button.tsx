import React, { FC, forwardRef, memo, HTMLProps } from 'react';
import './button.scss';

type ButtonVariant = 'primary' | 'secondary';
type ButtonSize = 'small' | 'large';

interface ButtonProps {
  size: ButtonSize;
  variant: ButtonVariant;
  className?: string;
  title?: string;
  disabled?: boolean;
  onClick?: () => void;
}

const Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
  const { size, variant, children, className = '', title, disabled = false, onClick } = props;

  const additionalClassNames = `${className} tc__btn__${size} tc__btn__${variant}`;

  return (
    <button ref={ref} className={`tc__btn ${additionalClassNames}`} title={title} disabled={disabled} onClick={onClick}>
      {children}
    </button>
  );
});

export default memo(Button as FC<ButtonProps & HTMLProps<HTMLButtonElement>>);
