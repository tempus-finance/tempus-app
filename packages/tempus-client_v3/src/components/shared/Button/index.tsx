import React, { ForwardedRef, forwardRef, memo, ReactNode } from 'react';

import './button.scss';

export interface ButtonProps {
  children: ReactNode;
  className?: string;
  title?: string;
  disabled?: boolean;
  onClick?: () => void;
}

const Button = forwardRef((props: ButtonProps, ref: ForwardedRef<HTMLButtonElement>) => {
  const { children, className = '', title, disabled = false, onClick } = props;

  return (
    <button ref={ref} className={`tc__btn ${className}`} title={title} disabled={disabled} onClick={onClick}>
      {children}
    </button>
  );
});

export default memo(Button);
