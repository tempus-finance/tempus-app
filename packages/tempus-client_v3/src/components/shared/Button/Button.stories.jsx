import React from 'react';

import Button from './Button';

export default {
  title: 'Button',
  component: Button,
  argTypes: {},
};

const style = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

const Template = args => (
  <div style={style}>
    <Button {...args.small}>My Button</Button>
    <div style={{ width: '20px' }} />
    <Button {...args.large}>My Button</Button>
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  small: {
    title: 'primary',
    size: 'small',
    variant: 'primary',
  },
  large: {
    title: 'primary',
    size: 'large',
    variant: 'primary',
  },
};

export const Secondary = Template.bind({});
Secondary.args = {
  small: {
    title: 'secondary',
    size: 'small',
    variant: 'secondary',
  },
  large: {
    title: 'secondary',
    size: 'large',
    variant: 'secondary',
  },
};
