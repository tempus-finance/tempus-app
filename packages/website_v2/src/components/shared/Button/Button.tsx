import { FC, memo, MouseEvent, PropsWithChildren, useCallback } from 'react';

import './Button.scss';

interface ButtonProps {
  className?: string;
  width?: string;
  height?: string;
  onClick?: (event: MouseEvent<HTMLDivElement>) => void;
}

const Button: FC<ButtonProps> = props => {
  const { className, width, height, onClick, children } = props;

  const handleClick = useCallback(
    (event: MouseEvent<HTMLDivElement>) => {
      onClick?.(event);
    },
    [onClick],
  );

  return (
    <div
      className={`tw__button__container ${className ?? ''}`}
      style={{ width, height }}
      onClick={handleClick}
      role="button"
      tabIndex={0}
    >
      {children}
    </div>
  );
};

Button.defaultProps = {
  width: '',
  height: '',
  onClick: undefined,
};

export default memo<PropsWithChildren<ButtonProps>>(Button);
