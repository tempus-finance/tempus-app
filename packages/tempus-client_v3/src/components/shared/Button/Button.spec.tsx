import { fireEvent, render } from '@testing-library/react';
import Button, { ButtonProps } from './Button';

const mockOnClick = jest.fn();

const defaultProps: ButtonProps = {
  title: 'my title',
  onClick: mockOnClick,
};

const subject = (props: ButtonProps) => render(<Button {...props}>My Button</Button>);

describe('Button', () => {
  it('renders a button with a title attribute', () => {
    const { queryByTitle, queryByText } = subject(defaultProps);

    const actualText = queryByText('My Button');
    const actualLabel = queryByTitle('my title');

    expect(actualText).not.toBeNull();
    expect(actualLabel).not.toBeNull();

    expect(actualText).toMatchSnapshot();
  });

  it('calls `onClick` when the button is clicked', () => {
    const { getByRole } = subject(defaultProps);

    const button = getByRole('button');
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
