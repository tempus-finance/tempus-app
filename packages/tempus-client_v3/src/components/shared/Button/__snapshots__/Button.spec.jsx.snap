// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button with a title attribute 1`] = `
<button
  class="tc__btn  tc__btn__large tc__btn__primary"
  title="my title"
>
  My Button
</button>
`;
